### ISSUE TITLE
Not asking for password

### ISSUE THREAD

**Issue by WereCatf** [gitahead/gitahead#229]
I have GitAhead 2.5.7 installed under Ubuntu 19.04 and I have cloned a remote git-repo using HTTPS. Alas, when I try to push commits, GitAhead just says it failed with an HTTP-error 401 and not even once tries to ask me for the credentials. My understanding is that it's supposed to ask for such, so something is wrong. Alas, GitAhead doesn't spit out any useful information as to what it could be.

**Comment by kas-luthor** [comment]
The problem is that GitAhead doesn't correctly understand when credential helpers report that they haven't found the requested credentials - in this case most likely GNOME keyring.
Until the fix is properly rolled out, just manually add them to the keyring.

**Comment by WereCatf** [comment]
@kas-luthor I have configured git to use git-credential-libsecret and the credentials are seemingly stored  correctly as I can see them there just fine with e.g. Seahorse. Pushing commits from CLI manually works as supposed, but GitAhead still ain't workin'.

**Comment by hackhaslam** [comment]
@kas-luthor Can you elaborate? I know that people are having issues with the credential helpers, but I haven't been able to recreate it or understand what the issues are.

**Comment by kas-luthor** [comment]
@WereCatf have you used the same protocol for the credentials?

@hackhaslam
What happened was that GitAhead correctly asked the credential helpers for data, but got an empty response since no credentials were available. GitCredential::get() still returns true, so no input dialog is opened.

**Comment by WereCatf** [comment]
@kas-luthor Yup.

**Comment by kas-luthor** [comment]
Does you password contain an equals sign?

**Comment by WereCatf** [comment]
@kas-luthor Nope.

**Comment by kas-luthor** [comment]
Could you run the git-credential-libsecret binary that resides in the same folder as GitAhead in a terminal with the parameter `get` and type the following:
```
protocol=https
host=<the domain of your origin URL>


```
(including two empty lines)

Does that output the correct credentials?

**Comment by WereCatf** [comment]
@kas-luthor Yes, it does. I checked with inotify that the binary is never called by GitAhead at all in the first place.

**Comment by kas-luthor** [comment]
That's odd - GitAhead should show the credential dialog when it's unable to start the helper executable.
Could you try to see if GitAhead calls git-credential-gnome-keyring?

**Comment by WereCatf** [comment]
@kas-luthor I actually checked that as well, I just forgot to mention that; neither executable gets called at all.

That said, I just came to think that is my issue because my git-server isn't Github, Bitbucket, Beanstalk or Gitlab? Does GitAhead only work properly with those? I mean, I don't understand why it would be like that, since git is git and not a single other application I've ever used has had any issues interacting with the server, but I guess anything could be possible.

**Comment by kas-luthor** [comment]
GitAhead works without problems with my Gitea server, so I'd say it does work with any proper GIT server. Which server software do you use? Is it possible to obtain a testing account on the instance you're using?

If no executable gets called, GitAhead should fall back to internal password caching, in which case a credential dialog should pop up.
I'm at my wits end here.

@hackhaslam do you have any ideas?

**Comment by WereCatf** [comment]
@kas-luthor I whipped you up an account, should've gotten an email about it. Lemme know when you don't need it anymore.

**Comment by kas-luthor** [comment]
Yep, got it. I'll look into it this afternoon when I'm home.

**Comment by hackhaslam** [comment]
I just released a new build. If it is one of the issues that @kas-luthor fixed, then it's available to test now.

**Comment by kas-luthor** [comment]
@WereCatf I wasn't able to reproduce the error.
If the latest release doesn't fix the error, try upgrading or downgrading your QT libs.

Other than that, I'm at my wits end.

I tested using a recently (last week) fully updated Arch Linux x86_64:
- QT 5.13.0-1
- Kernel 5.1.16-arch1-1-ARCH
- GCC 9.1.0
- ninja 1.9.0
- cmake 3.14.5

**Comment by tommai78101** [comment]
@kas-luthor Have you tried to suspend your Linux operating system, then wake the system up, before reattempting to do a `git push` in GitAhead?

I've been doing a lot of `pm-suspend` on Ubuntu, and GitAhead would fail to prompt you the password every time after re-awaking my machine.
